dd-sdk-android/src/test/kotlin/com/datadog/android/core/internal/persistence/file/batch/BatchFileDataReaderTest.kt

```diff
@@ -407,4 +397,18 @@ internal class BatchFileDataReaderTest {
     }
 
     // endregion
+
+    // region private
+
+    private fun List<ByteArray>.join(
```

Comment:
I've seen that you already created `Collection<ByteArray>.join` extension for this, this looks much more simpler though, wondering why not having the same implementation also there.

Reply:
Because the one used in test is not very efficient - it will create new array for every `reduce` operation, but it is much more readable indeed (this is an advantage for the test).

The one used in the production code creates array only once and is using `copyTo` to fill it.